content/automate/ManageIQ/System/CommonMethods/MiqAe.class/__methods__/weightedupdatestatus.rb

```diff
@@ -122,6 +136,11 @@ def main
                   # We set the task message.
                   if @handle.root['ae_state_step'] == 'on_error'
                     task.message = 'Failed'
+                    task.set_option('cleanup_request_id', create_cleanup_request.id)
```

Comment:
The `create_cleanup_request` method already returns an `id`.

Reply:
If `create_cleanup_request` does not go through the `if sm_uri.present?` block then the return will be `nil`.  Either way this method will currently error out trying to access `create_cleanup_request.id`.